src/controller/SetUpCodePairer.cpp

```diff
@@ -138,9 +138,23 @@ CHIP_ERROR SetUpCodePairer::StartDiscoverOverIP(SetupPayload & payload)
                                                       : Dnssd::DiscoveryFilterType::kLongDiscriminator;
     currentFilter.code =
         payload.isShortDiscriminator ? static_cast<uint16_t>((payload.discriminator >> 8) & 0x0F) : payload.discriminator;
+
+    // We're going to ensure that anything we discover matches currentFilter
```

Comment:
Not sure I get it. `SetUpCodePairer::NodeMatchesCurrentFilter` already validates if the service's `CM` key is different than 0, so we should ignore nodes that are not commissionable. If the node is able to update its PTR records, it can update the TXT records as well, so both ways of filtering should be equivalent in terms of their results and choosing what we query is only the matter of efficiency. Or is there something I'm not aware of?

Reply:
The issue is that when a node _updates_ the value of a TXT record, we don't have a good way of finding out about it: a browse operation of the sort we are doing does not notice that.  So we ignore the node initially because CM=0 in the TXT record, and then never notice that it becomes CM=1.

But when a node _adds_ a PTR record where the key is exact type we are browsing for (`_CM._sub_.matterc._udp`), we get notified.

Reply:
Updated comments to make this clearer.